Sources/Observer.swift

```diff
@@ -68,6 +84,15 @@ extension Signal {
 			}
 		}
 
+		deinit {
+			if interruptsOnDeinit {
+				// Since `Signal` would ensure that only one terminal event would ever be
+				// sent for any given `Signal`, we do not need to assert any condition
+				// here.
+				action(.interrupted)
```

Comment:
IMO `Observer` is part of the `Signal` contract. It also breaks the useful assumption of terminal events being sent at most once, that would potentially break things e.g. cleaning up resource twice, triggering cancellation code after successful completion.